service/src/main/java/bio/terra/pipelines/service/PipelinesService.java

```diff
@@ -41,14 +47,112 @@ public Pipeline getPipeline(PipelinesEnum pipelineName) {
+          String.format("Problems with pipelineInputs: %s", String.join("; ", errorMessages)));
+    }
+  }
+
+  private LinkedHashMap<String, Object> castInputsToMap(Object inputs) {
+    try {
+      return objectMapper.convertValue(inputs, LinkedHashMap.class);
+    } catch (IllegalArgumentException e) {
+      throw new ValidationException("pipelineInputs must be a JSON object");
+    }
+  }
+
+  /**
+   * Validate that all required inputs are present in the inputsMap
+   *
+   * @param inputDefinitions - list of input definitions for a pipeline
+   * @param inputsMap - map of inputs to validate
+   * @return list of egrror messages for missing required inputs
+   */
+  public List<String> validateRequiredInputs(
+      List<PipelineInputDefinition> inputDefinitions, Map<String, Object> inputsMap) {
+    ArrayList<String> errorMessages = new ArrayList<>();
+    inputDefinitions.stream()
+        .filter(PipelineInputDefinition::getIsRequired)
+        .forEach(
+            inputDefinition -> {
+              String inputName = inputDefinition.getName();
+              if (!(inputsMap.containsKey(inputName))) {
+                errorMessages.add(String.format("%s is required", inputName));
+              }
+            });
+    return errorMessages;
+  }
+
+  /**
+   * Validate that all present inputs are the correct type. We do not check for required inputs
+   * here.
+   *
+   * @param inputDefinitions - list of input definitions for a pipeline
+   * @param inputsMap - map of inputs to validate
+   * @return list of error messages for inputs that are not the correct type
+   */
+  public List<String> validateInputTypes(
+      List<PipelineInputDefinition> inputDefinitions, Map<String, Object> inputsMap) {
+    ArrayList<String> errorMessages = new ArrayList<>();
+    inputDefinitions.forEach(
+        inputDefinition -> {
+          String inputName = inputDefinition.getName();
+          if (inputsMap.containsKey(inputName)) {
+            PipelineInputTypesEnum inputType =
+                PipelineInputTypesEnum.valueOf(inputDefinition.getType().toUpperCase());
+            try {
+              inputType.cast(
+                  inputName, inputsMap.get(inputName)); // cast method includes a null check
+            } catch (ValidationException e) { // custom message from PipelineInputTypesEnum
+              errorMessages.add(e.getMessage());
+            } catch (IllegalArgumentException e) {
+              errorMessages.add(
+                  String.format(
+                      // note that for security we return the name of the field, not the
+                      // user-provided value
+                      "%s must be of type %s", inputName, inputDefinition.getType()));
+            }
+          }
+        });
+    return errorMessages;
+  }
+
+  /**
+   * Check for extra inputs that are not defined in the pipeline; log a warning if there are
+   * unexpected inputs
+   */
+  public void checkForExtraInputs(
```

Comment:
this method isn't tested because testing the logger is a pain but I checked that extra input does get logged at the warn level, like so:
```
2024-04-30T13:32:51.917-04:00 eRG9YDZX  WARN 729 --- [pipelines] [nio-8080-exec-1] b.t.pipelines.service.PipelinesService   : Extra inputs provided: extra_input
```